components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14321,4 +14322,167 @@ public void removeExpiredJWTs() throws APIManagementException {
+     * @param category     Category
+     * @return Category
+     */
+    public APICategory addCategory(int tenantID, APICategory category) throws APIManagementException {
+        String uuid = UUID.randomUUID().toString();
+        category.setId(uuid);
+        try (Connection connection = APIMgtDBUtil.getConnection();
+            PreparedStatement statement = connection.prepareStatement(SQLConstants.ADD_CATEGORY_SQL)) {
+            statement.setString(1, uuid);
+            statement.setString(2, category.getName());
+            statement.setString(3, category.getDescription());
+            statement.setInt(4, tenantID);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            handleException("Failed to add Category: " + uuid, e);
+        }
+        return category;
+    }
+
+    /**
+     * Update API Category
+     *
+     * @param apiCategory API category object with updated details
+     * @throws APIManagementException
+     */
+    public void updateCategory(APICategory apiCategory) throws APIManagementException {
+        try (Connection connection = APIMgtDBUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQLConstants.UPDATE_API_CATEGORY)) {
+            statement.setString(1, apiCategory.getDescription());
+            statement.setString(2, apiCategory.getName());
+            statement.setString(3, apiCategory.getId());
+            statement.execute();
+        } catch (SQLException e) {
+            handleException("Failed to update API Category : " + apiCategory.getName() + " of tenant " +
+                    APIUtil.getTenantDomainFromTenantId(apiCategory.getTenantID()), e);
+        }
+    }
+
+    /**
+     * Get all available API categories of the tenant
+     *
+     * @param tenantID
+     * @return API Categories List
+     */
+    public List<APICategory> getAllCategories(int tenantID) throws APIManagementException {
+        List<APICategory> categoriesList = new ArrayList<>();
+        try (Connection connection = APIMgtDBUtil.getConnection();
+            PreparedStatement statement = connection.prepareStatement(SQLConstants.GET_CATEGORIES_BY_TENANT_ID_SQL)) {
+            statement.setInt(1, tenantID);
+
+            ResultSet rs = statement.executeQuery();
+            while(rs.next()) {
+                String id = rs.getString("UUID");
+                String name = rs.getString("NAME");
+                String description = rs.getString("DESCRIPTION");
+
+                APICategory category = new APICategory();
+                category.setId(id);
+                category.setName(name);
+                category.setDescription(description);
+                category.setTenantID(tenantID);
+
+                categoriesList.add(category);
+            }
+        } catch (SQLException e) {
+            handleException("Failed to retrieve API categories for tenant " + tenantID, e);
+        }
+
```

Comment:
extra spaces in this method

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/8077/commits/e646d5a8cbf82d6fb2ba2cd090e252a4dfaca540